app/dojo.py

```diff
@@ -39,25 +40,65 @@ def create_room(self, rm_type, rm_names):
 				else:
 					return ("'room_name' should be a string value!")
 				
-	def add_person(self, name, role, wants_accomodation='N'):
+	def add_person(self, name, role, wants_accomodation="N"):
 		"""Add a new person to the Dojo and allocate 
 		either office space or living space 
 		"""
-		if role.lower() == "staff":
-			new_staff = Staff(name)
-			self.all_staff.append(new_staff)
-			print (" {} {} has been successfully added" .format(role, name))
-			self.allocate_room(new_staff)
-		elif role.lower() == "fellow":
-			new_fellow = Fellow(name, wants_accomodation)
-			self.all_fellows.append(new_fellow)
-			print ("{} {} has been successfully added" .format(rm_type, rm_name))
-			self.allocate_room(new_fellow)
-		else:
-			return ("You can only be staff or fellow at the Dojo")
-
-	def allocate_room(person):
-		pass
-	
+		if wants_accomodation.lower() != "Y" or wants_accomodation.lower() != "N":
+			return ("Invalid accomodation choice")
+		#if checks the role against accomodation option to allocate respective room
+		if wants_accomodation.lower() == "y":
+			if role.lower() == "fellow":
+				new_fellow = Fellow(name, wants_accomodation)
+				self.all_fellows.append(new_fellow)
+				return ("{} {} has been successfully added" .format(rm_type, rm_name))
+				self.allocate_available_officespace(new_fellow)
+				self.allocate_available_livingspace(new_fellow)
+			elif role.lower() == "staff":
+				return "Staff cannot be assigned Living Space"
+		elif wants_accomodation.lower() == "n":
+			if role.lower() == "staff":
+				new_staff = Staff(name)
+				self.all_staff.append(new_staff)
+				return (" {} {} has been successfully added" .format(role, name))
+				self.allocate_available_officespace(new_staff)
+			elif role.lower() == "fellow":
+				new_fellow = Fellow(name)
+				self.all_fellows.append(new_fellow)
+				return ("{} {} has been successfully added" .format(rm_type, rm_name))
+				self.allocate_available_officespace(new_fellow)
+				self.allocate_available_livingspace(new_fellow)
```

Comment:
If I remember the specification correctly, when a fellow opts for no accommodation they do not get any. I think this line is should be excluded.